app/domain/authentication/persist.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'command_class'
+
+module Authentication
+
+  module Default
+    # Performs additional setup for newly persisted authenticators
+    class InitializeDefaultAuth
+      extend CommandClass::Include
+      include AuthorizeResource
+
+      command_class(
+        dependencies: {
+          secret: Secret
+        },
+        inputs: %i[conjur_account service_id auth_data current_user]
+      ) do
+        def call
+          @auth_data&.parameters&.each do |key, value|
+            policy_branch = format("conjur/%s/%s", @auth_data.auth_name, @service_id)
+            variable_id = format("%s:variable:%s/%s", @conjur_account, policy_branch, key)
+
+            auth(@current_user, :update, Resource[variable_id])
+            @secret.create(resource_id: variable_id, value: value)
+          end
+        end
+      end
+
+    end
+  end
+
+  # Persists a new authenticator + webservice in Conjur
+  class PersistAuth
+    extend CommandClass::Include
+
+    command_class(
+      dependencies: {
+        logger: Rails.logger,
+        auth_initializer: Authentication::Default::InitializeDefaultAuth.new,
+        policy_loader: Policy::LoadPolicy.new,
+        auth_data_class: Authentication::AuthnK8s::K8sAuthenticatorData
+      },
+      inputs: %i[conjur_account service_id resource current_user client_ip request_data]
+    ) do
+      def call
+        auth_data = @auth_data_class.new(@request_data)
+        raise ArgumentError, auth_data.errors.full_messages unless auth_data.valid?
+
+        policy_details = initialize_auth_policy(
+          policy_loader: @policy_loader,
+          resource: @resource,
+          current_user: @current_user,
+          client_ip: @client_ip,
+          auth_policy: auth_policy(auth_data: auth_data, service_id: @service_id)
+        )
+
+        @auth_initializer.(conjur_account: @conjur_account, service_id: @service_id, auth_data: auth_data, current_user: @current_user)
+
+        policy_details
+      end
```

Comment:
Digging into this method, it looks like there's a fair amount of "method wrapping" going on (a method which wraps another single method). This increases cognitive complexity and makes the coder harder to read. What about pushing it all up into the `call` method with something like:

```ruby
def call
  auth_data = @auth_data_class.new(@request_data)
  raise ArgumentError, auth_data.errors.full_messages unless auth_data.valid?

  @auth_initializer.(conjur_account: @conjur_account, service_id: @service_id, auth_data: auth_data, current_user: @current_user)

  @policy_loader.(
    delete_permitted: false,
    action: :update,
    resource: @resource,
    policy_text: ApplicationController.renderer.render(
      template: "policies/#{auth_data.auth_name}",
      locals: { service_id: @service_id, auth_data: auth_data }
    ),
    current_user: @current_user,
    client_ip: @client_ip
  )
end
```

Reply:
I moved the logic out of the individual functions however I cannot change the order of the initializer and policy load calls because the initailizer relies on the policy being loaded